mapiproxy/servers/default/nspi/dcesrv_exchange_nsp.c

```diff
@@ -235,29 +235,74 @@ static void dcesrv_NspiUnbind(struct dcesrv_call_state *dce_call,
 
 	DCESRV_NSP_RETURN(r, 1, NULL);
 }
+
+static enum MAPISTATUS position_in_table(TALLOC_CTX *mem_ctx, struct STAT *pStat, 
+					 struct emsabp_context *emsabp_ctx, struct PropertyTagArray_r *mids,
+					 uint32_t *out_row, uint32_t *out_last_row)
+{
+	enum MAPISTATUS			retval;
+	uint32_t			row;
+	uint32_t			last_row;
+
+	if (mids->cValues > 0) {
+		last_row = mids->cValues - 1;
+	} else {
+		last_row = 0;
+	}
+        
+	if (pStat->CurrentRec == MID_CURRENT) {
+		/* Fractional positioning (3.1.4.5.2) */
+		row = pStat->NumPos * (last_row+1) / pStat->TotalRecs;
+		if (row > last_row) {
+			row = last_row;
+		}
+	} else {
+		/* Absolute positioning (3.1.4.5.1) */
+		if (pStat->CurrentRec == MID_BEGINNING_OF_TABLE) {
+			row = 0;
+		}
+		else if (pStat->CurrentRec == MID_END_OF_TABLE) {
+			row = last_row;
+		} else {
+			retval = MAPI_E_NOT_FOUND;
+			row = 0;
+			while (row <= last_row) {
+				if ((uint32_t) mids->aulPropTag[row] == (uint32_t) pStat->CurrentRec) {
```

Comment:
mids->aulPropTag[row]  is already an `uint32_t` (the other side could be int though)